src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -340,20 +340,22 @@ public String actualPathForPublicPath(String publicPath) {
             String lectureId = publicPath.replace(filename, "").replace("/api/files/attachments/lecture/", "");
             return Path.of(FilePathService.getLectureAttachmentFilePath(), lectureId, filename).toString();
         }
-        if (publicPath.contains("files/attachments/attachment-unit") && !publicPath.contains("/slide")) {
-            String attachmentUnitId = publicPath.replace(filename, "").replace("/api/files/attachments/attachment-unit/", "");
-            return Path.of(FilePathService.getAttachmentUnitFilePath(), attachmentUnitId, filename).toString();
-        }
-        if (publicPath.contains("files/attachments/attachment-unit") && publicPath.contains("/slide")) {
-            final var slideSubPath = publicPath.replace(filename, "").replace("/api/files/attachments/attachment-unit/", "").split("/");
-            final var shouldBeAttachmentUnitId = slideSubPath[0];
-            final var shouldBeSlideId = slideSubPath.length >= 3 ? slideSubPath[2] : null;
-            if (!NumberUtils.isCreatable(shouldBeAttachmentUnitId) || !NumberUtils.isCreatable(shouldBeSlideId)) {
-                throw new FilePathParsingException("Public path does not contain correct shouldBeAttachmentUnitId or shouldBeSlideId: " + publicPath);
+        if (publicPath.contains("files/attachments/attachment-unit")) {
+            if (!publicPath.contains("/slide")) {
```

Comment:
Sorry to nitpick again. Please do the positive value first if you have an if else anyway => Switch the conditional bodies.

Or, if you want to have the short one first, just extract the body of the else clause. As you return in the if clause that doesn't matter then, but you have less nesting.

Reply:
extracted else clause